Issue#21

GrzegorzDryja opened the pull request:
Check [last commit](https://github.com/tomkowy/notatnik_parser/commit/185cef0dd94c6b86f99cefc0bf457a4e0d990e5a).

Commit:
issue#22 first atempt

SaveFile function is asking user to input full path with file name.
I've made try catch exception for UnauthorizedAccessException but it doesn't return any statement to user if note is save were he wish or by default. Will work  on it next time.

Commit:
Update issue#22

Try catch in saveFile function works now with boolean return. User get the information that something went wrong. But he can't correct the path at this moment. Still working on it.

Commit:
Menu aded to Notepad

NotepadMenu works with menu similar to MainMenu that allows to choose between new note or old one (opening existing file doesn't works yet).

Commit:
issue#21

New way of checking user path. Added new functions, some modification to work on them later.

Commit:
Update NotepadMenu.cs

Just some later saved changes.

Commit:
Big update

More & more lines of code :) I realy like how it's works now (lots of testings). There are for sure some shortcomings e.g with SaveFile function. But I've got solution which is used in opening file. Can't wait for your judgement of my work.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,141 @@
+                    fs.Write(textNote, 0, textNote.Length);
+                }
+            }
+            catch (UnauthorizedAccessException)
```
why you catch this type of exception

GrzegorzDryja replied:
The story is simple... I got this error in IDE during tests, I mentioned that I thoroughly tested this code. Sometimes happens that you try to save something somewhere where you aren't allowed to save this.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,141 @@
+            string note = Console.ReadLine();
+            ConsoleKeyInfo typedKey = Console.ReadKey();
+
+            if (typedKey.Key == ConsoleKey.Enter && SaveFile(note))
```
saveFile shoudn't execute in if. You should execude saveFile method and then pass returned value to if statement. You can try at first ask about new path and check it and then ask about data for file. Now save method method saves the file and asks about path. It's two responsibilities

tomkowy commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,141 @@
+
+        private void OpenTextFile(string insertPathValue)
+        {
+            using (FileStream fs = File.Open(insertPathValue, FileMode.Open)) 
```
You don't check that file exist. Maybe someone deleted it.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,141 @@
+            }
+        }
+
+        private bool IsUserPathInputValid(string insertPathValue)
```
This method check input and open file. You should create a method to check input and open fil(this one you have :)) and use their separately in MenageUserPathInput

tomku-mar reviewed (changes requested):
it's very good. You started to separate responsibility

GrzegorzDryja commented:
Thank you for your review @tomkowy, I really appreciate your help which is invaluable. I will include your suggestions in next commit.

Commit:
Update NotepadMenu.cs

OpenTextFile use ManageUserPathInput, wich now only ask in do/while loop for path and with help of IsUserPathInputValid checks is it correct.

Commit:
Update NotepadMenu.cs

Changes. Next work on Exception and earlier path validation

Commit:
Update NotepadMenu.cs

User get information about error.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,207 @@
+                return true;
+            }
+
+            Console.WriteLine("Proszę wybrać numer z listy");
```
move this somewere else. You have two dependencies: check input and show messages.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,207 @@
+            {   
+                Console.WriteLine("Podaj nazwę pliku z rozszerzeniem:");                
+                insertNameValue = Console.ReadLine();
+                insertPathAndNameValue = pathName + '/' + insertNameValue;
```
check method String.Format and better, string interpolation ($"")

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,35 +5,207 @@
+        {
+            if(File.Exists(insertPathAndNameValue))
+            {                
+                Console.WriteLine("{0} - ta ścieżka jest niewłaściwa.", insertPathAndNameValue);
```
again, two dependencies. The methods's name tells us that this is a file check

tomku-mar reviewed (approved):
In general is good. I add some comments, but you don't need commit it :)
